src/components/Users/UserFilters.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+import { Input, Select, Button } from "semantic-ui-react";
+
+import { useUsers } from "./UsersContext";
+
+const optionsRole = [
+  { key: "all", text: "All", value: "all" },
+  { key: "doctor", text: "Doctor", value: "doctor" },
+  { key: "admin", text: "Admin", value: "admin" },
+];
+
+const optionsActive = [
+  { key: "all", text: "All", value: "all" },
+  { key: "active", text: "Active", value: "true" },
+  { key: "deactivated", text: "Deactivated", value: "false" },
+];
+
+const optionsVerified = [
+  { key: "all", text: "All", value: "all" },
+  { key: "verified", text: "Verified", value: "true" },
+  { key: "unverified", text: "Unverified", value: "false" },
+];
+
+const UserFilters = () => {
+  const { filters, handleApplyFilters: onApply } = useUsers();
+  const [search, setSearch] = useState(filters.search);
+  const [role, setRole] = useState(filters.role);
+  const [active, setActive] = useState(filters.active);
+  const [verified, setVerified] = useState(filters.verified);
+
+  const handleChangeSearch = (e, data) => {
+    setSearch(data.value);
+  };
+
+  const handleSubmit = () => {
+    onApply({
+      search, role, active, verified,
+    });
+  };
+
+  const handleChangeRole = (e, data) => {
+    if (data.value !== "all") { setRole(data.value); } else setRole("");
```

Comment:
this is an uncommon format. please do this:

```
if (data.value !== "all") {
    setRole(data.value);
} else {
    setRole("");
}
```

or

```
setRole(data.value !== "all" ? data.value : "");
```

Reply:
although you should not do this here. `setRole` should always set real state. What you can do is in the fetch function, prepare parameters to send. for example:

```
const params = {
    "role": filters.role !== "all" ? filters.role : undefined // if undefined it wont be send to the backend with axios which is a benefit
}
```